python/tk_multi_launchapp/launch_apps.py

```diff
@@ -0,0 +1,651 @@
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import re
+import sys
+
+import sgtk
+from sgtk import TankError
+
+from .prepare_apps import prepare_launch_for_engine
+
+shotgun_data = sgtk.platform.import_framework("tk-framework-shotgunutils", "shotgun_data")
+
+
+class BaseLauncher(object):
+    # documentation explaining how to reconfigure app paths
+    HELP_DOC_URL = "https://support.shotgunsoftware.com/entries/95443887#Setting%20up%20Application%20Paths"
+
+    def __init__(self):
+        # Retrieve the TK Application from the current bundle
+        self._tk_app = sgtk.platform.current_bundle()
+
+        # Store the current platform value
+        self._platform_name = {
+            "linux2": "linux", "darwin": "mac", "win32": "windows"
+        }[sys.platform]
+
+        # Initialize DCC information used during the launch process
+        self._set_app_info(None, None, None, None, None)
+
+    def _set_app_info(self, menu_name, app_engine, app_path, app_args, app_version):
+        """
+        Register information required to successfully launch the DCC
+
+        :param menu_name: Menu name displayed to launch this DCC.
+        :param app_engine: The TK engine associated with the DCC to be launched.
+        :param app_path: Full path to the DCC. May contain environment variables
+                         and/or the locally supported {version}, {v0}, {v1}, ...
+                         variables.
+        :param app_args: Args string to pass to the DCC at launch time.
+        :param app_version: Specific version of DCC to launch. Used to parse
+                            {version}, {v0}, {v1}, ... information from.
+        """
+        self._menu_name = menu_name
+        self._app_engine = app_engine
+        self._app_path = app_path
+        self._app_args = app_args
+        self._app_version = app_version
+
+    def _init_launch_command(self, app_menu_name, app_icon, app_engine, app_path, app_args, version=None):
+        """
+        Register a launch command with the current engine.
+
+        :param app_menu_name: Menu name to display to launch this DCC. This is
+                              also used to construct the associated command name.
+        :param app_icon: Icon to display for this DCC
+        :param app_engine: The TK engine associated with the DCC to be launched
+        :param app_path: Full path name to the DCC. This may contain environment
+                         variables and/or the locally supported {version}, {v0},
+                         {v1}, ... variables
+        :param app_args: Args string to pass to the DCC at launch time
+        :param version: (Optional) Specific version of DCC to use.
+        """
+        # Resolve the input path to the application to launch. Verify that
```

Comment:
it would be useful with some debug output at some point here, so that you can see details in the log around which entries have been registered.